tests/test_web_api.py

```diff
@@ -0,0 +1,307 @@
+
+        response = self.client.get("/rp/job/pipeline/", format="json")
+        self.assertEqual(0, len(response.data))
+
+    def test_create_and_read(self):
+        expected = self.test_create_ints()
+
+        url = urlparse(expected["url"])
+        response = self.client.get(url.path, format="json")
+        self._stock_response_check(expected, response.data)
+
+        response = self.client.get("/rp/job/pipeline/", format="json")
+        self.assertEqual(1, len(response.data))
+        self._stock_response_check(expected, response.data[0])
+
+    def test_create_delete_and_read(self):
+        expected = self.test_create_ints()
+
+        url = urlparse(expected["url"])
+        response = self.client.delete(url.path, format="json")
+        self.assertEqual(204, response.status_code)
+
+        response = self.client.get(url.path, format="json")
+        self.assertEqual(404, response.status_code)
+
+        response = self.client.get("/rp/job/pipeline/", format="json")
+        self.assertEqual(0, len(response.data))
+
+
+class RegulationFileTestCase(APITestCase):
+
+    file_contents = "123"
+    hashed_contents = None
+
+    def get_hashed_contents(self):
+        if self.hashed_contents is None:
+            try:
+                self.hashed_contents = md5(self.file_contents.encode(
+                    "utf-8")).hexdigest()
+            except AttributeError:
+                self.hashed_contents = md5(self.file_contents).hexdigest()
+        return self.hashed_contents
+
+    def test_create_file(self):
+        with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+            tmp.write(self.file_contents.encode("utf-8"))
+            tmp_name = ospath.split(tmp.name)[-1]
+            tmp.seek(0)
+            response = self.client.post(
+                "/rp/job/upload/", {"file": tmp})
+        self.assertEquals(201, response.status_code)
+        data = response.data
+        self.assertEquals(self.get_hashed_contents(), data["hexhash"])
+        self.assertEquals(tmp_name, data["filename"])
+        self.assertEquals("File contents not shown.", data["contents"])
+        self.assertEquals(file_url(self.get_hashed_contents()), data["url"])
+        return response
+
+    def test_reject_duplicates(self):
+        self.test_create_file()
+        with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+            tmp.write(self.file_contents.encode("utf-8"))
+            tmp.seek(0)
+            response = self.client.post(
+                "/rp/job/upload/", {"file": tmp})
+        self.assertEquals(400, response.status_code)
+        self.assertIn("error", response.data)
+        self.assertEquals("File already present.", response.data["error"])
+
+    def test_reject_large(self):
+        with patch("regparser.web.jobs.views.FileUploadView",
+                   new=PatchedFileUploadView) as p:
+            p.size_limit = 10000
+            with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
```

Comment:
For later: I think you'll really like py.test's [`tmpdir`](http://doc.pytest.org/en/latest/tmpdir.html) fixtures